iGCS/Managers/GCSDataManager.m

```diff
@@ -20,13 +23,45 @@ + (instancetype) sharedInstance {
     return sharedInstance;
 }
 
+
 - (instancetype) init {
     self = [super init];
     if (self) {
         _craft = [GCSCraftModelGenerator createInitialModel];
         _lastViewedMapCamera = nil;
+        _gcsSettings = [[GCSSettings alloc] init];  
     }
     return self;
 }
 
++ (instancetype)loadInstance {
+    NSData *decodedData = [NSData dataWithContentsOfFile:[GCSDataManager filePath]];
+    if (decodedData) {
+        GCSDataManager *progData = [NSKeyedUnarchiver unarchiveObjectWithData:decodedData];
+        return progData;
+    }
+    
+    return [[GCSDataManager alloc] init];
```

Comment:
@leskowicz 
You are just returning and empty GCSDataManager here. Instead create a method called loadArchives
1) try to read the data from disk and unarchive it.
2) if you have decoded data then do _gcsSettings = progData
3) If not then do _gcsSettings = [[GCSSettings alloc] init];

In init call:
[self loadArchives];

Probably will be more to do on this but see if that works for now.